File: Algorithms/111. Minimum Depth of Binary Tree/Solution.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public int minDepth(TreeNode root) {
        if (root==null) {
            return 0;
        }
        if(root.left==null&&root.right==null){
            return 1;
        }
        int left=-1;
        if(root.left!=null){
           left=minDepth(root.left)+1;
        }
        int right=-1;
        if(root.right!=null){
           right=minDepth(root.right)+1;
        }
        if(left==-1){
            return right;
        }
        if(right==-1){
            return left;
        }
    
        
        return Math.min(left, right);
    }
}

File: Algorithms/111. Minimum Depth of Binary Tree/Solution.rb
# Definition for a binary tree node.
# class TreeNode
#     attr_accessor :val, :left, :right
#     def initialize(val)
#         @val = val
#         @left, @right = nil, nil
#     end
# end

# @param {TreeNode} root
# @return {Integer}
def min_depth(root)
    if root
        left=root.left ? min_depth(root.left) : nil
        right=root.right ? min_depth(root.right) : nil
        left= left ? left : (right ? right : 0)
        right= right ? right : (left ? left : 0)
        return 1+[left,right].min()
    else
        return 0
    end
end